### PR TITLE
tests for chaospy.dist.backend.Dist class

### PR DESCRIPTION
**Pull request by davidovitch**
I'll start working on some test cases here, but it is still very much a work-in-progress.

### CODE REVIEW

**Comment by jonathf** [comment]
As I said, this would be much appreciated.

I've added a `work-in-progress` label. If you feel the PR is ready for review, swap it for the `ready to merge` label, and I will take a look.

File: src/chaospy/dist/collection.py [hunk 1]
```diff
@@ -223,18 +223,18 @@ def Exponpow(shape=0, scale=1, shift=0):
     return dist
 
 
-def Exponweibull(a=1, b=1, scale=1, shift=0):
+def Exponweibull(a=1, c=1, scale=1, shift=0):
```
Comment (davidovitch) [inline src/chaospy/dist/collection.py]: Is changing the API like this allowed? This will brake things for ```Exponweibull``` users...

**Comment by davidovitch** [comment]
@jonathf, not sure if it is ready to merge, but I would appreciate your opinion on the current state of affairs. Note that I've now changed the name of keyword argument of ```Exponweibull``` in order it to be consistent with the backend, and with the scipy.stats naming. But braking the API like this might be more trouble for users than it is worth anything. What do you think?

File: src/chaospy/dist/baseclass.py [hunk 2]
```diff
@@ -396,7 +396,7 @@ def __sub__(self, X):
 
     def __rsub__(self, X):
         """Y.__rsub__(X) <==> Y-X"""
-        return chaospy.dist.operators.add(X, -self)
+        return chaospy.dist.operators.add(self, -X)
```
Comment (davidovitch) [inline src/chaospy/dist/baseclass.py]: I am not familiar with operator overloading or how to do it correctly, but without these fixes only left subtraction/divition/etc works. This fixes at least the newly added tests in ```test_dist.py```

Comment (davidovitch) [inline src/chaospy/dist/baseclass.py]: This is is the same fix is used for the right addition fix in PR #26

**Comment by jonathf** [comment]
Sorry for late response.

I will take a look at your PR and come with some feedback this weekend.

File: src/chaospy/dist/baseclass.py [hunk 3]
```diff
@@ -416,31 +416,31 @@ def __div__(self, X):
...
+        return chaospy.dist.operators.mul(self, X**-1)
 
     def __pow__(self, X):
         """Y.__pow__(X) <==> Y**X"""
         return chaospy.dist.operators.pow(self, X)
 
     def __rpow__(self, X):
         """Y.__pow__(X) <==> X**Y"""
-        return chaospy.dist.operators.pow(X, self)
+        return chaospy.dist.operators.pow(self, X)
```
Comment (jonathf) [inline src/chaospy/dist/baseclass.py]: Following the text above, these seems incorrect.

The doc in `__rpow__` is wrong. Feel free to fix it.

File: src/chaospy/dist/cores.py [hunk 4]
```diff
@@ -857,8 +857,8 @@ def _cdf(self, x, a, c):
         return (exm1c)**a
     def _ppf(self, q, a, c):
         return (-np.log1p(-q**(1.0/a)))**(1.0/c)
-    def _bnd(self, c):
-        return 0, self._ppf(1-1e-10, c)
+    def _bnd(self, a, c):
+        return 0, self._ppf(1-1e-10, a, c)
```
Comment (jonathf) [inline src/chaospy/dist/cores.py]: Thanks for this.

**Comment by davidovitch** [comment]
Thanks for the input! I hope to get another stab at it later this week or during the weekend.

**Comment by jonathf** [comment]
Sorry. I haven't looked at this in a while.

I see you have updated the code as I suggested. The only thing worth noting is that you have a few tests commented out. Does this mean you are still working on them, or should they perhaps be removed before I complete the merge?

**Comment by davidovitch** [comment]
No problem, I wanted to consider a few more extra test cases before pinging you again, but maybe I could remove the current commented tests, and leave more tests for another PR? Otherwise this might take too long...

**Comment by jonathf** [comment]
Going for more than one PR is fine.
Remove the commented section, ping me, and I will merge this.

**Comment by davidovitch** [comment]
@jonathf ready for merging (I don't think I can change the label to "ready to merge")

**Comment by jonathf** [comment]
Thanks for this.

**Pull request merged**